src/packages/board/board.py

```diff
@@ -172,68 +220,79 @@ def __delitem__(self, key: _KEY_TYPE):
-
-        automatically assigns
+        Context manager to modify a Rescue
 
         Args:
-            pass through to :class:Rescue 's constructor
-
-            *args (): args to pass to :class:`Rescue` 's constructor
-            overwite(bool): overwrite existing rescues
-            **kwargs (): keyword arguments to pass to Rescue's constructor
+            key ():
 
         Yields:
-            created rescue object
+            Rescue: rescue to modify based on its `key`
         """
 
-        rescue = Rescue(*args, board_index=self.free_case_number, **kwargs)
-        yield rescue
-        # then append it to ourselves
-        await self.append(rescue, overwrite=ovewrite)
+        target = self[key]
 
-    async def append(self, rescue: Rescue, overwrite: bool = False) -> None:
-        """
-        Append a rescue to ourselves
+        # most tracked attributes may be modified in here, so we pop the rescue
+        # from tracking and append it after
 
+        del self[key]
 
+        try:
+            # Yield so the caller can modify the rescue
+            yield target
 
-        Args:
-            rescue (Rescue): object to append
-            overwrite(bool): overwrite existing cases
-    """
-        if (rescue.api_id in self or rescue.board_index in self) and not overwrite:
-            raise ValueError("Attempted to append a rescue that already exists to the board")
-        self._storage_by_uuid[rescue.api_id] = rescue
-        self._storage_by_index[rescue.board_index] = rescue
+        finally:
+            # we need to be sure to re-append the rescue upon completion
+            # (so errors don't drop cases)
+            await self.append(target)
 
-        if rescue.client:
-            self._storage_by_client[rescue.client] = rescue
+        # If we are in online mode, emit update event to API.
+        if self.online:
+            logger.trace("updating API...")
+            await self._handler.update_rescue(target)
 
     @asynccontextmanager
-    async def modify_rescue(self, key: _KEY_TYPE) -> Rescue:
+    async def create_rescue(self, *args, ovewrite=False, **kwargs) -> Rescue:
         """
-        Context manager to modify a Rescue
+        Context manager that creates a rescue on the board for use
+
+        automatically assigns
 
         Args:
-            key ():
+            pass through to :class:Rescue 's constructor
+
+            *args (): args to pass to :class:`Rescue` 's constructor
+            overwite(bool): overwrite existing rescues
+            **kwargs (): keyword arguments to pass to Rescue's constructor
 
         Yields:
-            Rescue: rescue to modify based on its `key`
+            created rescue object
         """
+        index = self.free_case_number
+        rescue = Rescue(*args, board_index=index, **kwargs)
 
-        target = self[key]
+        try:
+            if not self.online:
+                logger.warning("creating case in offline mode...")
```

Comment:
Log output tense.  "Created case in offline mode"

Reply:
No, this log event occurs prior to the rescue being created on the API, thus present tense is correct.
Especially since this action can result in a failure, past-tense is inappropriate.